src/image-featurizer/src/test/scala/ImageFeaturizerSuite.scala

```diff
@@ -34,6 +34,32 @@ class ImageFeaturizerSuite extends LinuxOnly with CNTKTestUtils with RoundTripTe
     compareToTestModel(result)
   }
 
+  test("structured streaming"){
+
+    val model = new ImageFeaturizer()
+      .setInputCol("image")
+      .setOutputCol(outputCol)
+      .setModelLocation(session, s"${sys.env("DATASETS_HOME")}/CNTKModel/ConvNet_CIFAR10.model")
+      .setCutOutputLayers(0)
+      .setLayerNames(Array("z"))
+
+    val imageDF = session
+      .readStream
+      .format(classOf[ImageFileFormat].getName)
+      .schema(ImageReader.imageDFSchema)
+      .load(cifarDirectory)
+
+    val resultDF = model.transform(imageDF)
+
+    val q1 = resultDF.writeStream
+      .format("memory")
+      .queryName("images")
+      .start()
+
+    Thread.sleep(3000)
```

Comment:
Any way to check for 6 images being found more directly than wait for 3 seconds? What if it took 4 seconds sometimes or 1 second (in which case you'll be blocking for longer than needed).